src/spark/utils.py

```diff
@@ -113,6 +114,14 @@ def get_spark_session(
 
     spark_conf = _get_base_spark_conf(app_name, executor_cores, yarn)
     sc = {}
+    if os.environ.get('USE_KUBE_SPAWNER') == 'true':
+        yarn = False  # YARN is not used in the Kubernetes spawner
+        # Since the Spark driver cannot resolve a pod's hostname without a dedicated service for each user pod,
+        # use the pod IP as the identifier for the Spark driver host
+        hostname = os.environ["SPARK_DRIVER_HOST"]
+        if hostname:
```

Comment:
Should this throw an error if there's no host?

Reply:
I think that's fine. In kubespawner, this would never happen. `SPARK_DRIVER_HOST` is set in kubespawner.

Reply:
In that case, it seems like if there's no host name something has gone Really Wrong and throwing an error would be the right thing to do